promdens.py

```diff
@@ -554,14 +564,16 @@ def windowing(self):
         arr_print[1:, :] = self.weights
 
         np.savetxt('pdaw.dat', arr_print.T, fmt=['%8d'] + ['%16.5e']*self.nstates,
-                   header=f"Convolution: '{self.conv}'\nParameters:  fwhm = {self.field_fwhm/self.fstoau:.3f} fs, "
-                          f"t0 = {self.field_t0/self.fstoau:.3f} fs\n"
+                   header=f"Convolution: '{self.conv}'\nParameters:  fwhm = {self.field_params.fwhm/self.fstoau:.3f} fs, "
+                          f"t0 = {self.field_params.t0/self.fstoau:.3f} fs\n"
                           f"index        " + str(' '*8).join([f"weight S{s + 1:d}" for s in range(self.nstates)]))
 
         print("  - Weights saved to file 'pdaw.dat'.")
 
 
 def plot_spectrum(ics: InitialConditions) -> None:
+    import matplotlib.pyplot as plt
```

Comment:
If we turn on plotting, it will be importing matplotlib 3 or 4 times. Won't it slow down the code importing it multiple times?

Reply:
Good question. Python "caches" imports, so the import cost is only paid once. In fact I've specifically moved the import inside these functions, so that for example if you just print help, it will now be much snapier.

Reply:
Cool, I didn't know it imports only once. I'm very happy with this change, it's indeed unnecessary to import matplotlib when we don't plot.